ownCloud/Messages/MessageGroupCell.swift

```diff
@@ -0,0 +1,248 @@
+
+		// Remove old views
+		alertView?.removeFromSuperview()
+		alertView = nil
+
+		applyAllContainer?.removeFromSuperview()
+
+		containerView?.removeFromSuperview()
+
+		guard let message = messageGroup.messages.first else {
+			return
+		}
+
+		let messages = messageGroup.messages
+		let multiMessage = messages.count > 1
+
+		if let choices = message.choices {
+			for choice in choices {
+				let option = AlertOption(label: choice.label, type: choice.type, handler: { [weak self] (_, _) in
+					if let applyAllSwitch = self?.applyAllSwitch, applyAllSwitch.isOn {
+						if let messages = self?.messageGroup?.messages {
+							for message in messages {
+								if let messageChoice = message.choice(withIdentifier: choice.identifier) {
+									queue.resolveMessage(message, with: messageChoice)
+								}
+							}
+						}
+					} else {
+						queue.resolveMessage(message, with: choice)
+					}
+				})
+
+				options.append(option)
+			}
+		}
+
+		if let title = message.localizedTitle, let description = message.localizedDescription {
+			if containerView == nil {
+				containerView = UIView()
+				containerView?.translatesAutoresizingMaskIntoConstraints = false
+			}
+
+			alertView = AlertView(localizedTitle: title, localizedDescription: description, options: options)
+			alertView?.translatesAutoresizingMaskIntoConstraints = false
+
+			containerView?.layer.cornerRadius = 10
+			containerView?.layer.masksToBounds = true
+			containerView?.backgroundColor = UIColor(white: 0.70, alpha: 0.15)
+
+			if multiMessage {
+				var setupLayout : Bool = false
+
+				if applyAllContainer == nil {
+					applyAllContainer = UIView()
+					applyAllContainer?.translatesAutoresizingMaskIntoConstraints = false
+					applyAllContainer?.backgroundColor = UIColor(white: 0.60, alpha: 0.10)
+
+					setupLayout = true
+				}
+
+				if applyAllSwitch == nil {
+					applyAllSwitch = UISwitch()
+					applyAllSwitch?.translatesAutoresizingMaskIntoConstraints = false
+					applyAllSwitch?.accessibilityLabel = "Apply choice to all similar issues".localized
+					applyAllSwitch?.addTarget(self, action: #selector(applyAllSwitchChanged), for: .primaryActionTriggered)
+
+					applyAllContainer?.addSubview(applyAllSwitch!)
+				}
+
+				if applyAllLabel == nil {
+					applyAllLabel = UILabel()
+					applyAllLabel?.translatesAutoresizingMaskIntoConstraints = false
+					applyAllLabel?.font = UIFont.systemFont(ofSize: UIFont.systemFontSize)
+					applyAllLabel?.text = "Apply to all".localized
+
+					applyAllContainer?.addSubview(applyAllLabel!)
+				}
+
+				if badgeLabel == nil {
+					badgeLabel = RoundedLabel(text: "", style: .token)
+					badgeLabel?.translatesAutoresizingMaskIntoConstraints = false
+				}
+
+				badgeLabel?.isHidden = true
+
+				if showAllLabel == nil {
+					showAllLabel = UIButton(type: .system)
+					showAllLabel?.translatesAutoresizingMaskIntoConstraints = false
+					showAllLabel?.setTitle("Show all", for: .normal)
```

Comment:
`localize` is missing, be sure this string is available in localization file

Reply:
Good catch! Fixed.